Src/AutoFixture/FreezeOnMatchCustomization.cs

```diff
@@ -64,6 +66,44 @@ public FreezeOnMatchCustomization(Type targetType)
         }
 
         /// <summary>
+        /// Initializes a new instance of the <see cref="FreezeOnMatchCustomization"/> class.
+        /// </summary>
+        /// <param name="parameter">
+        /// The <see cref="ParameterInfo"/> used to freeze specimens decorated with advanced
+        /// attributes (e.g. <see cref="StringLengthAttribute"/>).
+        /// </param>
+        /// <param name="matcher">
+        /// The <see cref="IRequestSpecification"/> used to match the requests
+        /// that will be satisfied by the frozen specimen.
+        /// </param>
+        /// <exception cref="ArgumentNullException">
+        /// <paramref name="parameter"/> or <paramref name="matcher"/> is null.
+        /// </exception>
+        public FreezeOnMatchCustomization(
+            ParameterInfo parameter,
```

Comment:
Well, yes, sort of. I'm predictable, it seems :neutral_face: 

Sorry to be a bit negative, but I think this needs to go in a slightly different direction. The overall idea, however, seems sound :+1: 

Adding a `ParameterInfo` property pulls in the direction of less cohesion, which I'm not particularly fond of. Now the class has two properties, both of which could be `null`, or both of which, according to the type system, could have a value. This is the sort of feature extension that tends to rot a code base over time.

Some time ago, @ecampidoglio and I had [a discussion about something similar](https://github.com/AutoFixture/AutoFixture/pull/317#discussion_r19058880) that might be worth revisiting. Essentially, the idea is to widen the existing 'greedy' constructor, so that it takes any `object`, instead of only a `Type`. AFAICT, there's nothing in the class's invariants that require it to be a `Type`.

Given that we now seem to have a use for this change, I think it might be worthwhile pursuing.